dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -482,10 +482,15 @@ public String toString() {
         public Supplier<Channel> stickyChannels() {
             return () -> this;
         }
+
+        @Override
+        public EndpointChannel endpoint(Endpoint _endpoint) {
+            return _request -> Futures.immediateFailedFuture(exceptionSupplier.get());
+        }
     }
 
     /* Abstracts away DialogueChannel so that we can handle no-service/no-uri case in #getInternalDialogueChannel. */
-    private interface InternalDialogueChannel extends Channel {
+    private interface InternalDialogueChannel extends Channel, EndpointChannelFactory {
```

Comment:
This is the most important change, we have a lot of code which attempts to cast Channel to EndpointChannelFactory and use `endpoint(endpoint)` on the result, failing that it will call the `Channel.execute(Endpoint,Request)` method, which internally delegates to the preferred `endpoint` method. Unfortunately that ends up re-computing a great deal of info on each invocation.

This cruft exists to avoid breaking dialogue API :-/